src/org/zaproxy/zap/extension/pscanrulesAlpha/XAspNetVersionScanner.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2017 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+
+import net.htmlparser.jericho.Source;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Plugin.AlertThreshold;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+
+/**
+ * a scanner to passively scan for the presence of the server HTTP response header
+ */
+public class XAspNetVersionScanner extends PluginPassiveScanner {
+
+    /**
+     * Prefix for internationalised messages used by this rule
+     */
+    private static final String MESSAGE_PREFIX = "pscanrules.xaspnetversioncanner.";
+
+    private PassiveScanThread parent = null;
+
+    private final List<String> xAspNetHeaders = new ArrayList<String>();
+
+    public XAspNetVersionScanner() {
+        xAspNetHeaders.add("X-AspNet-Version");
+        xAspNetHeaders.add("X-AspNetMvc-Version");
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        if (AlertThreshold.LOW.equals(this.getLevel())) {
```

Comment:
It should be the opposite, in `LOW` it should raise more issues not less:
https://github.com/zaproxy/zap-core-help/wiki/HelpUiDialogsOptionsPscanrules

Reply:
I am thinking of removing it and always raise an alert, what do you think? This can expose the version of ASP.Net...

Reply:
Sounds good, the other "header leak" scanners also raise always.